.ci/scripts/coverity.sh

```diff
@@ -0,0 +1,139 @@
+#!/bin/bash -eE
+# shellcheck disable=SC2086,SC2181
+set -o pipefail
+
+handle_error() {
+    err_code=$?
+    local script_name="${0##*/}"
+    local last_command="${BASH_COMMAND}"
+    local error_line="${BASH_LINENO[${BASH_LINENO[@]} - 1]}"
+    echo "Error: ${script_name} - Command '${last_command}' failed at line ${error_line}."
+    exit $err_code
+}
+trap 'handle_error' ERR
+
+#
+# This script runs static code analysis using Coverity.
+# The script needs to be run from the main directory.
+#
+export UCC_PASSWORD=$UCC_PASSWORD
+export UCC_USERNAME=$UCC_USERNAME
+topdir=$(git rev-parse --show-toplevel)
+cd "$topdir" || exit 1
+./autogen.sh
+./configure
+module load tools/cov-2021.12
+nproc=$(grep -c processor /proc/cpuinfo)
+make_opt="-j$((nproc / 2 + 1))"
```

Comment:
`nproc` is a utility by itself. 
Just use $(nproc), no need to define it beforehand.

Reply:
removed

Reply:
it raise a SC https://www.shellcheck.net/wiki/SC2154
even if nproc is exist

Reply:
Try
`make_opt="-j$(($(nproc) / 2 + 1))"`

Reply:
Yes, Alexey's variant is proper.
You cannot auto substitute command output without `$(<cmd>)` operator in the math operator `$((<math_calculations>))`.
`nproc` is a standard utility.

Wrong:
`make_opt="-j$((nproc / 2 + 1))"` - wrong

Will work, but bad approach (you redefine the standard cmd without a need):
```
nproc=$(grep -c processor /proc/cpuinfo)
make_opt="-j$((nproc / 2 + 1))"
```

Best option:
`make_opt="-j$(($(nproc) / 2 + 1))"`

Reply:
changed to best option